satpy/utils.py

```diff
@@ -208,31 +209,42 @@ def proj_units_to_meters(proj_str):
 
 
 def _get_sunz_corr_li_and_shibata(cos_zen):
+    return 24.35 / (2. * cos_zen + xu.sqrt(498.5225 * cos_zen**2 + 1))
 
-    return 24.35 / (2. * cos_zen +
-                    xu.sqrt(498.5225 * cos_zen**2 + 1))
 
-
-def sunzen_corr_cos(data, cos_zen, limit=88.):
+def sunzen_corr_cos(data, cos_zen, limit=88., max_sza=95.):
     """Perform Sun zenith angle correction.
 
     The correction is based on the provided cosine of the zenith
-    angle (*cos_zen*).  The correction is limited
-    to *limit* degrees (default: 88.0 degrees).  For larger zenith
-    angles, the correction is the same as at the *limit*.  Both *data*
-    and *cos_zen* are given as 2-dimensional Numpy arrays or Numpy
-    MaskedArrays, and they should have equal shapes.
+    angle (``cos_zen``).  The correction is limited
+    to ``limit`` degrees (default: 88.0 degrees).  For larger zenith
+    angles, the correction is the same as at the ``limit`` if ``max_sza``
+    is `None`. The default behavior is to gradually reduce the correction
+    past ``limit`` degrees up to ``max_sza`` where the correction becomes
+    0. Both ``data`` and ``cos_zen`` should be 2D arrays of the same shape.
 
     """
 
     # Convert the zenith angle limit to cosine of zenith angle
-    limit = xu.cos(xu.deg2rad(limit))
+    limit_rad = np.deg2rad(limit)
```

Comment:
I assume `xu` and `np` are equivalent here, since which version of xarray and numpy ? Is it something we need to indicate in setup.py ?

Reply:
http://xarray.pydata.org/en/stable/api.html#universal-functions

Mainly:

```
With recent versions of numpy, dask and xarray, NumPy ufuncs are now supported directly on all xarray and dask objects. This obliviates the need for the xarray.ufuncs module, which should not be used for new code unless compatibility with versions of NumPy prior to v1.13 is required.
```

Reply:
So we need to fix setup.py to read `numpy >= 1.13.0` I suppose

Reply:
Yes, I suppose you are right.